sound/soc/sof/intel/hda-pcm.c

```diff
@@ -95,6 +95,83 @@ static inline u32 get_bits(struct snd_sof_dev *sdev, int sample_bits)
 	}
 };
 
+/* There are two dma modes in dsp:
+ * (1) host dma and link dma work in decouple mode for SOF + HDA
+ * (2) host dma and GP dma work in couple mode for SOF + I2S
+ * The difference is that in decopule mode, the host dma is set by dai
+ * frontend and link dma is set by dai backend and the setting maybe
+ * different while in couple mode, host dma and GP dma share the same
+ * dma setting.
+ */
```

Comment:
No. you are using the term 'coupled' mode is an overloaded way. In the I2S mode, the Host and GP DMA have nothing to do with each other, and likewise in HDaudio mode there should be no dependency.
I don't see why you introduce new confusing concepts

Reply:
yes, it is exactly overloaded. I just want to make it clear but now confusing, I will refine it

Reply:
yeah, "coupled" to me means there is a dependency relationship between the two DMACs meaning you have the meaning incorrect for 1 & 2 above. i.e. there is no coupling with between Host GW and GP DMA. 

Can you tell name the DSP registers that are used in binding, it's not obvious from the HAS.

Reply:
yes, it is not obvious from the HAS. I just did some tests to get it.
In programming guide, the register is: PPLCxCTL